libraries/AP_HAL_ChibiOS/hwdef/CUAV-7-Nano/hwdef.dat

```diff
@@ -0,0 +1,295 @@
+
+# board ID for firmware load
+APJ_BOARD_ID 7000
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# to be compatible with the px4 bootloader we need
+# to use a different RAM_MAP
+env USE_ALT_RAM_MAP 1
+
+# flash size
+FLASH_SIZE_KB 2048
+
+# with 2M flash we can afford to optimize for speed
+env OPTIMIZE -O2
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 UART7 UART5 USART1 UART8 USART3 OTG2
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+PA9  VBUS INPUT OPENDRAIN
+
+# pins for SWD debugging
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# telem1
+PE8  UART7_TX UART7
+PF6  UART7_RX UART7
+PF8  UART7_RTS UART7
+PE10 UART7_CTS UART7
+
+# telem2
+PC8  UART5_RTS UART5
+PC9  UART5_CTS UART5
+PC12 UART5_TX UART5
+PD2  UART5_RX UART5
+
+# GPS1
+PB6  USART1_TX USART1
+PB7  USART1_RX USART1
+
+# GPS2
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# debug uart
+PD8  USART3_TX USART3
+PD9  USART3_RX USART3
+
+# uart6, RX only, RC input, if no IOMCU
+PC7  USART6_RX USART6
+
+# ethernet (not implemented yet)
+PC1  ETH_MDC            ETH1
+PA2  ETH_MDIO           ETH1
+PC4  ETH_RMII_RXD0      ETH1
+PC5  ETH_RMII_RXD1      ETH1
+PG13 ETH_RMII_TXD0      ETH1
+PG12 ETH_RMII_TXD1      ETH1
+PB11 ETH_RMII_TX_EN     ETH1
+PA7  ETH_RMII_CRS_DV    ETH1
+PA1  ETH_RMII_REF_CLK   ETH1
+#PG15 ETH_POWER_EN      ETH1
+
+PG15 Ethernet_PWR_EN OUTPUT HIGH # disable power on ethernet
+
+define BOARD_PHY_ID  MII_LAN8742A_ID
+define BOARD_PHY_RMII
+
+# ADC
+PA0  SCALED1_V3V3           ADC1 SCALE(2)
+PA4  SCALED2_V3V3           ADC1 SCALE(2)
+PB1  VDD_5V_SENS            ADC1 SCALE(2)
+PC0  RSSI_IN                ADC1 SCALE(1)
+PF12 FMU_SERVORAIL_VCC_SENS ADC1 SCALE(3)
+
+# analog in
+PC2  BATT_VOLTAGE_SENS ADC1 SCALE(1) ANALOG(9)
+PB0  BATT_CURRENT_SENS ADC1 SCALE(1) ANALOG(8)
+
+# pin7 on AD&IO, analog 12
+PF3 ADC3_6V6 ADC3 SCALE(2) ANALOG(12)
+
+# pin6 on AD&IO, analog 13
+PC3 ADC1_3V3 ADC1 SCALE(1) ANALOG(13)
+
+define HAL_BATT_VOLT_PIN 9
+define HAL_BATT_CURR_PIN 8
+define HAL_BATT_VOLT_SCALE 10.1
+define HAL_BATT_CURR_SCALE 17.0
+
+# SPI1 - IIM42652
+PA5  SPI1_SCK SPI1
+PB5  SPI1_MOSI SPI1
+PG9  SPI1_MISO SPI1
+PI9  IIM42652_CS CS
+PF2  IIM42652_DRDY INPUT
+
+# SPI2 - BMI088
+PI1  SPI2_SCK SPI2
+PI2  SPI2_MISO SPI2
+PI3  SPI2_MOSI SPI2
+PH5  BMI088_A_CS CS
+PG2  BMI088_G_CS CS
+PG3  BMI088_DRDY_GYR INPUT
+PA10 BMI088_DRDY_ACC INPUT
+
+# SPI4 - BMP581
+PE12 SPI4_SCK SPI4
+PE13 SPI4_MISO SPI4
+PE14 SPI4_MOSI SPI4
+PD15 BMP581_CS CS
+PG1  BMP581_DRDY INPUT
+
+# SPI5 - FRAM
+PF7  SPI5_SCK  SPI5
+PH7  SPI5_MISO SPI5
+PF11 SPI5_MOSI SPI5
+PG7  FRAM_CS CS
+
+# SPI devices 
+SPIDEV bmi088_g SPI2 DEVID1  BMI088_G_CS    MODE3 10*MHZ  10*MHZ
+SPIDEV bmi088_a SPI2 DEVID2  BMI088_A_CS    MODE3 10*MHZ  10*MHZ
+SPIDEV iim42652 SPI1 DEVID1  IIM42652_CS    MODE3  2*MHZ   8*MHZ
+SPIDEV ramtron  SPI5 DEVID1  FRAM_CS        MODE3  8*MHZ   8*MHZ
+SPIDEV bmp581   SPI4 DEVID1  BMP581_CS      MODE3 7.5*MHZ 12*MHZ
+
+# PWM output pins
+PI0  TIM5_CH4  TIM5  PWM(1)  GPIO(50)
```

Comment:
It's an H7, I would highly recommend at least enabling bdshot on the first four motors.

Reply:
Shouldn't we enable bdshot in another suffixed `-bdshot` hwdef file? We're testing for enabling the bdshot on the first four motors. If got no problem, we will update this commit.

Reply:
There's no reason to unless you really want it - I try and make sure that new flight controllers have bdshot support out of the box